s2cloudless/sentinelhub_masking.py

```diff
@@ -30,41 +34,38 @@ class CloudMaskRequest:
 
     def __init__(
         self,
-        cloud_detector,
-        bbox,
-        time,
+        cloud_detector: S2PixelCloudDetector,
+        bbox: BBox,
+        time: Union[
+            str,
+            Tuple[str, str],
+            dt.date,
+            Tuple[dt.date, dt.date],
+            dt.datetime,
+            Tuple[dt.datetime, dt.datetime],
+            List[dt.datetime],
+        ],
```

Comment:
`dt.datetime` already covers `dt.date` . In any case, if you have a multi-line type definition, it's best to define the type as a constant at the top of the file.
we also have a `sentinelhub.types.RawTimeType` or something similar, check if it fits here. There is also a type for a time interval.

Reply:
updated